### ISSUE TITLE
VS Code extension: Add IntelliSense suggestions in debug console

### ISSUE THREAD

**Issue by dougbu** [OmniSharp/omnisharp-vscode#299]
Nothing currently assists users as they type out expressions in the VS Code debug console. Not sure if there's already a feature the C# extension could hook into. But whether or not VS Code needs a new feature as well, this would be a good improvement to the debugging story.

BTW the ability to expand objects in the debug console output window helps but isn't sufficient. For one thing, method calls aren't supported there.

**Comment by gregg-miskelly** [comment]
@dougbu can you open this in https://github.com/Microsoft/vscode/issues?

**Comment by dougbu** [comment]
@gregg-miskelly is work needed here as well as there i.e. should this issue remain open too? What's the VS Code term for IntelliSense?

**Comment by gregg-miskelly** [comment]
I would guess that the IntelliSense interface for this new feature would have to be slightly different than the current interactions between VS Code and extensions since the expression is typed in a different physical document.

**Comment by dougbu** [comment]
Filed Microsoft/vscode#6263

**Comment by gregg-miskelly** [comment]
> Is anything else needed to get this to work in the C# extension?

Yes, there is a whole lot of work in the C# extension required to support this. This is not a trivial feature, as the way the feature works in VS Code is really designed for dynamic languages (ex: JavaScript). Statically typed languages need to build a bring from the debugger back to the language service for this feature to work.

**Comment by terryaney** [comment]
@gregg-miskelly So are you saying we shouldn't expect intellisense to work in Debug or Watch windows?  Because mine aren't.

**Comment by gregg-miskelly** [comment]
@terryaney This issue tracks adding support to the C# extension for supporting intellisense in any of the debugger evaluation windows. However, VS Code doesn't support intellisense in the watch window. So to have it in the watch window, the C# team needs to complete this feature, and the VS Code team needs to complete https://github.com/microsoft/vscode/issues/48810.

**Comment by MyPledge** [comment]
Any updates?
